### PR TITLE
feat: gas phase reaction scheme 

### PR DESCRIPTION
**Pull request by adambasha0**

### CODE REVIEW

File: app/api/chemotion/reaction_api.rb [hunk 1]
```diff
@@ -164,6 +164,8 @@ class ReactionAPI < Grape::API
         optional :segments, type: Array
         optional :variations, type: [Hash]
         optional :vessel_size, type: Hash
+        optional :gaseous, type: Boolean
+        optional :vessel_size, type: Hash
```
Comment (JanCBrammer) [inline app/api/chemotion/reaction_api.rb:168]: It looks like there are duplicate changes from rebasing `reaction-scheme-adaptation` onto `main` (with `main` containing #2017). See for example this duplicate of `:vessel_size`.

**Review by JanCBrammer — CHANGES_REQUESTED** [review]

File: app/api/chemotion/reaction_api.rb (hunk 1 above)
Comment (adambasha0) [reply app/api/chemotion/reaction_api.rb]: fixed

File: spec/usecases/reactions/update_materials_spec.rb [hunk 2]
```diff
@@ -120,5 +135,19 @@
         expect(Sample.find_by(name: 'product').target_amount_value).to eq(99.08304)
       end
     end
+
+    context 'when vessel volume is valid' do
+      it 'returns the calculated mole gas product' do
+        result = class_instance.send(:update_mole_gas_product, product_sample, 80)
+        expect(result).to eq(3.5543368129550247e-06)
```
Comment (JanCBrammer) [inline spec/usecases/reactions/update_materials_spec.rb:142]: Is this high of a precision relevant? If not, maybe reduce precision to reduce potential for this test to break.

File: spec/javascripts/utils/UnitConversion.spec.js
Comment (JanCBrammer) [inline spec/javascripts/utils/UnitConversion.spec.js:1]: Maybe test some corner cases as well? What happens when passing values that the function does not expect, e.g., non-numeric values, negative values? Some resilience might be important when `UnitConversion` serves as a re-usable module.

**Review by JanCBrammer — CHANGES_REQUESTED** [review]
- From a UI perspective, positioning the toggle button ("Gas Scheme" to "Default Scheme" and vice versa) inside the scheme tab isn't ideal. have to be careful to not toggle the state when I'm switching tabs.
- Conceptually, what's the difference between a catalyst reference ("CAT") and a feedstock reference ("FES")? Can the material marked as "Ref" be both? If only the material marked as "Ref" can be "FES", the UI needs to be adapted accordingly. Currently, when marking another material as "Ref", "FES" isn't adapted accordingly.

**Review by JanCBrammer — COMMENTED** [review]
Could 6cd6cba4a4ebfa91872943a40f2a79233f8a4fed (fixing flaky chemical API tests) be a separate PR?
Would be good to have this in `main` asap.

File: app/packs/src/apps/mydb/elements/details/reactions/ReactionDetailsMainProperties.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ import { statusOptions } from 'src/components/staticDropdownOptions/options';
 import LineChartContainer from 'src/components/lineChart/LineChartContainer';
 import EditableTable from 'src/components/lineChart/EditableTable';
 import { permitOn } from 'src/components/common/uis';
+import GaseousReactionActions from 'src/stores/alt/actions/GaseousReactionActions';
```
[the remark this answers is not in the document]
Comment (adambasha0) [reply app/packs/src/apps/mydb/elements/details/reactions/ReactionDetailsMainProperties.js]: done

**Comment by adambasha0** [comment]
> * From a UI perspective, positioning the toggle button ("Gas Scheme" to "Default Scheme" and vice versa) inside the scheme tab isn't ideal. I have to be careful to not toggle the state when I'm switching tabs.
> * Conceptually, what's the difference between a catalyst reference ("CAT") and a feedstock reference ("FES")? Can the material marked as "Ref" be both? If only the material marked as "Ref" can be "FES", the UI needs to be adapted accordingly. Currently, when marking another material as "Ref", "FES" isn't adapted accordingly.

1. As for UI, after a discussion with Nicole and others, it was decided to keep it as it is for now. 
2. There is no relation between Ref and CAT and FES, a REF material can be both. CAT and FES are also references in a gas phase reaction, which are used in some calculations:

Calculation1: Mol feedstock = Purity x 1 x V/(0.0821  x 294)
Calculation2: Mol Product = ppm x 1 x V (reaction vessel_size)/ (0.0821 x temp_in_K x 1000000)
Calculation3: TON = Mol gas product/mol CAT reference 
Calculation4: TOF = TON/time
Yield = Mol gas product / Mol feedstock ( Purity x 1 x V (reaction vessel_size)/(0.0821  x 294)) 
For more info, you can check [the requirements document](https://docs.google.com/document/d/1Jd6V5zqRv0cfw_xAJE-oHDYzSuBMQyBBz3wQQBxRGJo/edit)

**Comment by adambasha0** [comment]
> 1. it was decided to keep it as it is for now

done

File: app/packs/src/components/common/ToggleButton.js
[the remark this answers is not in the document]
Comment (adambasha0) [reply app/packs/src/components/common/ToggleButton.js:1]: You are right. ToggleButton component was supposed to be reusable. I refactored the component and now it is reusable

File: spec/usecases/reactions/update_materials_spec.rb (hunk 2 above)
Comment (adambasha0) [reply spec/usecases/reactions/update_materials_spec.rb:142]: This is related to a legacy code of the NumeralInputWithUnitsCompo component which needs to be refactored in a separate PR.

**Comment by JanCBrammer** [comment]
Is the feature documented over at https://github.com/ComPlat/chemotion_saurus?

**Comment by adambasha0** [comment]
[Gas Reaction Scheme feature:](https://github.com/user-attachments/assets/46f25ee0-753f-46b5-a15c-7836a34408dd)

File: app/packs/src/components/common/ToggleButton.js
[the remark this answers is not in the document]
Comment (JanCBrammer) [reply app/packs/src/components/common/ToggleButton.js:1]: Looks good.

File: spec/usecases/reactions/update_materials_spec.rb (hunk 2 above)
Comment (JanCBrammer) [reply spec/usecases/reactions/update_materials_spec.rb:177]: Is the refactoring necessary to reduce the numerical precision in the test?

Comment (adambasha0) [reply spec/usecases/reactions/update_materials_spec.rb:177]: yes, I guess so

File: spec/javascripts/utils/UnitConversion.spec.js
Comment (adambasha0) [reply spec/javascripts/utils/UnitConversion.spec.js:1]: could not agree more. done
